### DISCUSSION TITLE
How to force WPA3 connection

### DISCUSSION THREAD

**Discussion by Andbarz** [arendst/Tasmota#20859] [Q&A] [answered]
Hi everyone,
on my router I can select either WPA2 only or WPA2/WPA3 encryption. All my devices automatically use wpa3 (including esphome devices) except for the tasmota devices.

As far as I can tell tasmota now supports WPA3. How can I force tasmota to use that rather than the unsecure WPA2? Maybe that should be set as default

**Comment by Jason2866** [comment, marked as answer]
Tasmota does support WPA2 and WPA3. There is no API in Arduino to force or set to use WPA3 only.
The connection how it is done is not under control of Tasmota. The only way is to force it via settings in the router / AP.
esphome uses a different SDK (IDF not Arduino) where it is possible to set this.

**Comment by Andbarz** [comment]
Isn't it possible to just set wpa3 as default then? I'm not interested in
explicitly choosing wpa3, I'd just like that too be the default choice if
available.

Is it still something impossible due to the Arduino SDK?


Il lun 4 mar 2024, 12:59 Jason2866 ***@***.***> ha scritto:

**Comment by Jason2866** [comment]
To allow only WPA3 would lock out many devicess from users having equipment not supporting WPA3.
Beside of this, as already answered, no it is not possible with Arduino to set such options.

The solution is, swap your Router / APs against ones which allow more granular control ;-)

**Comment by Andbarz** [comment]
Don't misunderstand me, the following reply is just meant to clarify what I
was saying and not to complain/insist.

Obviously enabling only wpa3 on all tasmota firmwares doesn't make any
sense. I was just asking if it possible to automatically choose wpa3 if
that's advertised as available (or to try to connect with wpa3 and then
wpa2, I don't know how a WiFi connection is setup).
I don't know anything about how tasmota works, in my mind it could be
something as simple as changing a default value in some config file when
compiling the firmware. It that's not case I'll do something else.


Il lun 4 mar 2024, 19:56 Jason2866 ***@***.***> ha scritto:

**Comment by Jason2866** [comment]
This is not possible with Tasmota. The used SDK (Arduino) which is used for Tasmota has no API for this.
